pkg/util/util.go

```diff
@@ -23,3 +25,82 @@ func GetCertPool(paths []string) (*x509.CertPool, error) {
 	}
 	return pool, nil
 }
+
+func IsEndpointAllowed(endpoint *url.URL, allowedDomains []string) bool {
```

Comment:
This is a great refactor! Thanks!

Let's slide it to a more specific `utils` area though.

Can we move this to here: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/validation/utils.go

Since this is all validation related?

Reply:
Yes sure, done.